controllers/droptailer_controller.go

```diff
@@ -170,28 +166,17 @@ func (r *DroptailerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	if certificateBase == "" {
 		r.certificateBase = certificateBase
 	}
-
-	genericPredicate := predicate.Funcs{
-		GenericFunc: func(e event.GenericEvent) bool {
-			return e.Meta.GetNamespace() == namespace
-		},
-	}
-
-	mapToDroptailerReconcilation := handler.ToRequestsFunc(
-		func(a handler.MapObject) []reconcile.Request {
-			return []reconcile.Request{
-				{NamespacedName: types.NamespacedName{
-					Name:      "trigger-reconcilation-for-droptailer",
-					Namespace: namespace,
-				}},
-			}
-		})
-	triggerDroptailerReconcilation := &handler.EnqueueRequestsFromMapFunc{
-		ToRequests: mapToDroptailerReconcilation,
-	}
+	triggerDroptailerReconcilation := handler.EnqueueRequestsFromMapFunc(func(a client.Object) []reconcile.Request {
+		return []reconcile.Request{
+			{NamespacedName: types.NamespacedName{
+				Name:      "trigger-reconcilation-for-droptailer",
+				Namespace: namespace,
+			}},
+		}
+	})
 
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&corev1.Pod{}, builder.WithPredicates(genericPredicate)).
+		For(&corev1.Pod{}).
```

Comment:
unsure if this is expressed correct

Reply:
Purpose of predicates is to filter requests. Specifically in this case we filtered out only Generic events. And the rest of events were filtered out by the same criteria(known namespace) in reconcilation logic. So probably this change breaks nothing.

Still, i would suggest to remove this logic from reconcilation and create a predicate for all types of events. As this is recommended approach for filtering requests in controller-runtime.

Reply:
OK thanks, how can this be specified ? Link to sample would be much appreciated

Reply:
I pushed changes)

Reply:
Thanks!